src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+        if denoising_steps is not None and denoising_steps < 1:
+            raise ValueError("`denoising_steps` must be positive.")
+        if ensemble_size is not None:
+            if ensemble_size < 1:
+                raise ValueError("`ensemble_size` must be positive.")
+            if ensemble_size == 2:
+                logger.warning(
+                    "`ensemble_size` == 2 results are similar to no ensembling (1); "
+                    "consider increasing the value to at least 3."
+                )
+        if processing_resolution is not None:
+            if processing_resolution < 0:
+                raise ValueError(
+                    "`processing_resolution` must be non-negative: 0 for native resolution, "
+                    "or any positive value for downsampled processing."
+                )
+            if processing_resolution % self.latent_size_scale != 0:
+                raise ValueError(f"`processing_resolution` must be a multiple of {self.latent_size_scale}.")
+        if resample_method_input not in ("nearest", "nearest-exact", "bilinear", "bicubic", "area"):
+            raise ValueError(
+                "`resample_method_input` takes string values compatible with PIL library: "
+                "nearest, nearest-exact, bilinear, bicubic, area."
+            )
+        if resample_method_output not in ("nearest", "nearest-exact", "bilinear", "bicubic", "area"):
+            raise ValueError(
+                "`resample_method_output` takes string values compatible with PIL library: "
+                "nearest, nearest-exact, bilinear, bicubic, area."
+            )
+        if batch_size < 0:
+            raise ValueError("`batch_size` must be non-negative: 0 for auto-detection or any positive value.")
+        if output_prediction_format not in ["pt", "np", "pil"]:
+            raise ValueError("`output_prediction_format` must be one of `pt`, `np`, or `pil`.")
+        if input_latent is not None and generator is not None:
+            raise ValueError("`input_latent` and `generator` are cannot be used together.")
+        if ensembling_kwargs is not None and not isinstance(ensembling_kwargs, dict):
+            raise ValueError("`ensembling_kwargs` must be a dictionary.")
+        if output_visualization_kwargs is not None and not isinstance(output_visualization_kwargs, dict):
+            raise ValueError("`output_visualization_kwargs` must be a dictionary.")
+
+        # memory saving hints
+        if save_memory:
+            logger.warning(
+                f"`save_memory` is currently not implemented. Consider setting `batch_size` to the minimum (1), "
+                f"ensure that `processing_resolution` is set to the recommended default value "
+                f"({self.optimal_processing_resolution}). Further memory savings can be achieved with pipeline "
+                f"offloading. Refer to https://huggingface.co/docs/diffusers/optimization/memory for more information."
+            )
```

Comment:
This is an anti-pattern. Instead of a single `save_memory` argument in a pipeline call, we document ways for users to optimize the memory requirements.

Reply:
I agree; we used to have a `save_memory` argument in the community pipeline. Let's discuss what happens to that one and then I will clean up these deprecations

Reply:
I will also need guidance on implementing offloading. I'm not sure how it is supposed to work when we have a loop over the input images, and when for each image we execute the VAE encode, UNet, and VAE decode. I marked the calls to the modules with the `# Model invocation` comment

Reply:
Fixed